lib/lambda_ethereum_consensus/p2p/incoming_requests/handler.ex

```diff
@@ -128,6 +128,35 @@ defmodule LambdaEthereumConsensus.P2P.IncomingRequests.Handler do
     end
   end
 
+  defp handle_req("beacon_blocks_by_root/2/ssz_snappy", message_id, message) do
+    with <<24, snappy_blocks_by_root_request::binary>> <- message,
+         {:ok, ssz_blocks_by_root_request} <- Snappy.decompress(snappy_blocks_by_root_request),
+         {:ok, blocks_by_root_request} <-
+           Ssz.from_ssz(ssz_blocks_by_root_request, SszTypes.BeaconBlocksByRootRequest) do
+      ## TODO: there should be check that the `start_slot` is not older than the `oldest_slot_with_block`
+      %SszTypes.BeaconBlocksByRootRequest{body: body} =
+        blocks_by_root_request
+
+      count = length(body)
+
+      count = min(count, ChainSpec.get("MAX_REQUEST_BLOCKS"))
+
+      "[Received BlocksByRoot Request] requested #{count} number of blocks"
+      |> Logger.info()
+
+      blocks =
+        body
+        |> Enum.slice(0..(count - 1))
+        |> Enum.map(&BlockStore.get_block/1)
+
+      response_chunk =
+        blocks
+        |> Enum.map_join(&create_block_response_chunk/1)
```

Comment:
Nit: let's merge this two pipes
```suggestion
      response_chunk =
        body
        |> Enum.slice(0..(count - 1))
        |> Enum.map(&BlockStore.get_block/1)
        |> Enum.map_join(&create_block_response_chunk/1)
```